sbws/lib/relaylist.py

```diff
@@ -415,28 +422,49 @@ def _init_relays(self):
         # Change to stem.descriptor.remote in future refactor.
         network_statuses = c.get_network_statuses()
         new_relays_dict = dict([(r.fingerprint, r) for r in network_statuses])
+        log.debug("Number of relays in the current consensus: %s.",
+                  len(new_relays_dict))
 
         # Find the timestamp of the last consensus.
         timestamp = valid_after_from_network_statuses(network_statuses)
         self._consensus_timestamps.append(timestamp)
         self._remove_old_consensus_timestamps()
-        # Update the relays that were in the previous consensus with the
-        # new timestamp
+
         new_relays = []
+
+        # Only or debugging, count the relays that are not in the current
+        # consensus and have not been seen in the last consensuses either.
+        num_old_relays = 0
+
         relays = copy.deepcopy(self._relays)
         for r in relays:
             if r.fingerprint in new_relays_dict.keys():
```

Comment:
I know this change is not for this patch, but one can do `if key in dict` without first generating a list of keys and check for an item in that list.

Reply:
hmm, we're generating 2 new things here: the dictionary to search for keys and an a copy of the list of relays. While the later isn't really needed and could just be modified, the former is useful to be able to easily search in a dictionary.
Ah, or you mean that it could be `if r.fingerprint in new_relays_dict` without the last `.keys()`

Reply:
It's much faster if you don't use .keys().

Searching a list takes a time that depends on the length of the list (and the position of the relay in the list). That could be a long time for a big list of relays.

Searching a dict usually takes a shorter amount of time, because it's a hash table.